### PR TITLE
Opensearch

### PR DESCRIPTION
**Pull request by MichaelLangbein**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/dlr-eoc/ukis-frontend-libraries/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] The Project is [building and passes all tests](https://github.com/dlr-eoc/ukis-frontend-libraries/blob/master/DEVELOPMENT.md#further-you-can-test-and-build-locally) without errors

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
This merge-request summarizes the work done during the flood-system sprint.

Issue Number: N/A

## What is the new behavior?
Two new clients have been added to `services-ogc`:
 - `CswClient`: a javascript-client implementing the [CSW specification](https://www.ogc.org/standards/cat)
 - `OpensearchWrapperService`: a wrapper around [EOXC opensearch-client](https://www.npmjs.com/package/opensearch-browser). It exposes the clients URL-parsing mechanism to allow the end user to search not only `results`-links, but also `collection` links.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
All changes affect only `services-ogc`.

### CODE REVIEW

File: projects/services-ogc/src/lib/opensearch/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+## Native support
+
+Most browser have a method `window.external.AddSearchProvider` that speaks opensearch.
```
Comment (boeckMt) [inline projects/services-ogc/src/lib/opensearch/README.md:3]: Window.external is **Deprecated** - https://developer.mozilla.org/en-US/docs/Web/API/Window/external

File: projects/services-ogc/tsconfig.spec.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
     "types": [
       "jasmine",
       "node"
-    ]
+    ],
```
Comment (boeckMt) [inline projects/services-ogc/tsconfig.spec.json:8]: Do not use trailing commas in JSON.

File: package.json [hunk 3]
```diff
@@ -98,6 +99,7 @@
     "karma-jasmine": "~3.3.0",
     "karma-jasmine-html-reporter": "^1.5.0",
     "karma-junit-reporter": "^1.2.0",
+    "license-checker": "^25.0.1",
```
Comment (boeckMt) [inline package.json:102]: Why did you add license-checker, there are no changes in the demo for license or an automated use at buildtime.

Comment (MichaelLangbein) [reply package.json]: That's indeed superfluous. Removed! Thanks!

File: projects/services-ogc/tsconfig.spec.json (hunk 2 above)
Comment (MichaelLangbein) [reply projects/services-ogc/tsconfig.spec.json]: done

File: projects/services-ogc/src/lib/opensearch/README.md (hunk 1 above)
Comment (MichaelLangbein) [reply projects/services-ogc/src/lib/opensearch/README.md]: Thanks for the link! Removed.

**Comment by boeckMt** [comment]
@MichaelLangbein i added some comments to your PR.

On the next time please try to stick to our [Commit Message Guidelines](https://github.com/dlr-eoc/ukis-frontend-libraries/blob/master/CONTRIBUTING.md#commit) so in case we want to use a automated Changelog/Release in some time it can be created from the keywords.

**Comment by MichaelLangbein** [comment]
> @MichaelLangbein i added some comments to your PR.

Have I missed any of those comments? I've addressed all comments visible in this conversation. Anything else left to do? I'm more than happy to add any required additional changes!

**Review by boeckMt — APPROVED** [review]

**Comment by MichaelLangbein** [comment]
Awesome, thanks!
